crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -8,14 +8,44 @@ import "./interfaces/ICommonBridge.sol";
 /// @title CommonBridge contract.
 /// @author LambdaClass
 contract CommonBridge is ICommonBridge, Ownable, ReentrancyGuard {
-    constructor(address owner) Ownable(owner) {}
+    /// @notice Mapping of unclaimed withdrawals. A withdrawal is unclaimed if
+    /// there is a non-zero value in the mapping (a merkle root) for the hash
+    /// of the L2 transaction that requested the withdrawal.
+    /// @dev The key is the hash of the L2 transaction that requested the
+    /// withdrawal.
+    /// @dev The value is a boolean indicating if the withdrawal was claimed or not.
+    mapping(bytes32 => bool) public claimedWithdrawals;
+
+    /// @notice Mapping of merkle roots to the L2 withdrawal transaction logs.
+    /// @dev The key is the L2 block number where the logs were emitted.
+    /// @dev The value is the merkle root of the logs.
+    /// @dev If there exist a merkle root for a given block number it means
+    /// that the logs were published on L1, and that that block was committed.
+    mapping(uint256 => bytes32) public blockWithdrawalsLogs;
+
+    address public ON_CHAIN_PROPOSER;
 
-    struct WithdrawalData {
-        address to;
-        uint256 amount;
+    modifier onlyOnChainProposer() {
+        require(
+            msg.sender == ON_CHAIN_PROPOSER,
+            "CommonBridge: caller is not the OnChainProposer"
+        );
+        _;
     }
 
-    mapping(bytes32 l2TxHash => WithdrawalData) public pendingWithdrawals;
+    constructor(address owner) Ownable(owner) {}
+
+    function initialize(address onChainProposer) public nonReentrant {
```

Comment:
Should we limit this function to be called only once?
Maybe adding a `require(ON_CHAIN_PROPOSER == 0)` so once it's changed it cannot be changed anymore

Reply:
Yes! Makes sense.

Reply:
Done [285f41c](https://github.com/lambdaclass/lambda_ethereum_rust/pull/944/commits/285f41c8326b1899788ad0b8313477741f5313ca)